### PR TITLE
enable LX pass by default

### PR DESCRIPTION
**Pull request by chichun-charlie-liu**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [X] bug
- [X] feature
- [X] documentation
- [ ] cleanup

#### What this PR does:

1. turn on LX planning pass by default, ENV VAR `LX_PLANNING=1` is no longer needed. 
2. **An adjustment in `per_core_view` behavior to return `empty_view` (which will cause fallback to HBM) when it cannot decide the view accurately, as opposed to the current behavior using assertion fail to stop running right away** 
3. update documents that mentioned `LX_PLANNING=1` (only the main ones, there are still many instances to update, need a follow up PR)
4. minor adjustment to work-div/LX co-opt function (which is still guarded by ENV VAR, not turning on in this PR)
5. fix several bugs in per_core_view:
    - incorrect lookup table when there are duplicated strides
    - fix the logic when dividing "host stick-dim"
    - return empty_view if `buf` does not have `FixedTiledLayout`

**Results by `spyre-perf-suite`**
- when forced to turn off LX (using ENV VAR), i.e. the case before this PR.
```bash
LX_PLANNING=0 python run_benchmark.py --op softmax --shape 1 512 4096

Op: softmax  Shape: [[1, 512, 4096]]
Metric                             torch-spyre          sendnn tsp/sendnn
----------------------------------------------------------------------
wall_clock_ms.mean_ms                  4.544           4.510     1.008
cpu_ms.mean_ms                         4.439           4.458     0.996
spyre_ms.mean_ms                       0.449           1.107     0.406
kernel_ms.mean_ms                      0.305           0.097     **3.144**
memory_transfer_ms.mean_ms             0.145           1.009     0.144
pt_util%                               0.000           0.000        N/A
```
- when run by default
```bash
python run_benchmark.py --op softmax --shape 1 512 4096

Op: softmax  Shape: [[1, 512, 4096]]
Metric                             torch-spyre          sendnn tsp/sendnn
----------------------------------------------------------------------
wall_clock_ms.mean_ms                  5.537           4.116     1.345
cpu_ms.mean_ms                         5.433           4.062     1.338
spyre_ms.mean_ms                       0.373           1.009     0.370
kernel_ms.mean_ms                      0.104           0.098     **1.061**
memory_transfer_ms.mean_ms             0.268           0.911     0.294
pt_util%                               0.000           0.000        N/A
```

#### Which issue(s) this PR is related to:

#235 #2414 
#2166

### CODE REVIEW

**Review by ani300 — DISMISSED** [review]

File: torch_spyre/_inductor/config.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 from torch.utils._config_module import install_config_module
 
-lx_planning: bool = os.environ.get("LX_PLANNING", "0") == "1"
+lx_planning: bool = os.environ.get("LX_PLANNING", "1") == "1"
```
Comment (isuruf) [inline torch_spyre/_inductor/config.py:21]: You could do the following if you want to get rid of the environment variable.
```suggestion
lx_planning: bool = True
```

Comment (dgrove-oss) [reply torch_spyre/_inductor/config.py:21]: Agreed.  We want the config to be able to do measurements.  Just want to default it to True instead of False.

File: torch_spyre/_inductor/passes.py [hunk 2]
```diff
@@ -279,13 +279,12 @@ def __call__(self, graph: GraphLowering) -> None:
...
         cost_model_ops = cost_model_matmul_division(operations)
         work_distribution(operations, cost_model_ops)
-        if config.lx_planning:
-            allocator = (
-                StrategyBCoOptimizingAllocator()
-                if config.co_optimizing_lx_planning
-                else None
-            )
-            scratchpad_planning(graph, allocator=allocator)
+        allocator = (
```
Comment (dgrove-oss) [inline torch_spyre/_inductor/passes.py]: We should keep the config guard here.  Otherwise flipping the config to false doesn't actually do anything.

**Review by dgrove-oss — DISMISSED** [review]

File: torch_spyre/_inductor/passes.py (hunk 2 above)
Comment (chichun-charlie-liu) [reply torch_spyre/_inductor/passes.py]: fixed, thx

**Review by dgrove-oss — DISMISSED** [review]
Approved.  Looks good.

File: CLAUDE.md [hunk 3]
```diff
@@ -42,7 +42,6 @@ Test sub-suites:
 |---|---|
 | `TORCH_SPYRE_DEBUG=1` | Enable C++ debug logging and `-O0` builds |
 | `SENCORES` | Number of Spyre cores (1–32, default 32) |
-| `LX_PLANNING=1` | Enable LX scratchpad memory planning |
```
Comment (dgrove-oss) [inline CLAUDE.md:45]: Please revert this change too.  LX_PLANNING is still a supported env var

Comment (chichun-charlie-liu) [reply CLAUDE.md:45]: fixed

**Review by dgrove-oss — APPROVED** [review]
